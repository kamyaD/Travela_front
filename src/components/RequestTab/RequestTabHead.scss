  .request__tab {
    margin-top: 57px;
    padding: 0;
    margin: 0;
    list-style: none;
    display: flow-root;
  }


   .request__tab-card {
    box-sizing: border-box;
    min-height: 150px;
    width: 19.49%;
    border-radius: 16px;
    background-color: #FFFFFF;
    margin-right: 7.34%;
    float: left;
    position: relative;
    padding-bottom: 23px;
    &:nth-child(4n) {
      margin-right: 0px;
    }
    .tab-logo {
      width: 56px;
      height: 56px;
      border-radius: 50%;
      margin-top: 20px;
      margin-left: auto;
      margin-right: auto;
      background: #f5f5f5;
      svg {
        left: 50%;
        width: 29px;
        transform: translate(-50%, -50%);
        height: 19px;
        float: left;
        top: 50%;
        position: relative; 

       }
    }
    .tab-title {
      width: 100%;
      text-align: center;
      margin-top: 10px;
      height: 21px;
      color: #4F4F4F;
      font-family: "DIN Pro";
      font-size: 16px;
      font-weight: 500;
      line-height: 21px;
    }
    .foot-text {
      margin-top: 8px;
      height: 15px;
      width: 100%;
      color: #4F4F4F;
      font-family: "DIN Pro";
      font-size: 12px;
      line-height: 15px;
      text-align: center;
    }
  }


   .current {
    border: 1px solid #3359DB;
    opacity: 1 !important;
    .foot-text {
        display: block!important;
    }
  }


   .complete {
    opacity: 1 !important;
    .tab-logo {
      background-color: rgba(16, 163, 109, 0.15);
    }
    .mark {
      display: block;
    }
  }

   .incomplete {
    box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.1);
    opacity: 0.5;
    .foot-text {
      display: none;
    }
  }

   .mark {
    display: none;
    position: absolute;
    right: 10px;
    top: 10px;
    height: 24px;
    width: 24px;
    background-color: #10A36D;
    border-radius: 50%;
    text-align: center;

   }

   .new-request_title{
    height: 23px;	
    width: 277px;	
    color: #4F4F4F;	
    font-family: "DIN Pro";	
    font-size: 18px;	
    font-weight: 700;	
    line-height: 23px;
    margin-top:60px;
    margin-bottom: 25px;
  }

