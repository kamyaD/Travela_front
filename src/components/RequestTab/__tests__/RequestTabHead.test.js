import React from 'react';
import { shallow } from 'enzyme';
import RequestTabHead from '../RequestTabHead';
import tabIcons from '../../../images/icons/new-request-icons';

const props = {
  steps:[
    { id:1, name:'Personal Information', status:'', icon: tabIcons.personal },
    { id:2, name:'Trip Details', status:'', icon: tabIcons.tripDetails },
    { id:3, name:'Travel Stipends', status:'', icon: tabIcons.stipend },
    { id:4, name:'Travel Checklist', status:'', icon: tabIcons.checkList }   
  ],
  currentTab: 3,
};

let wrapper;
describe('<RequestTabHead />', () => {
  beforeEach(() => {
    wrapper=  shallow (<RequestTabHead {...props} />);
  });
  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('render all the step', () =>{
    expect(wrapper.find('.request__tab-card').length).toEqual(4);
  });
  it('calls renderTab', () => {
    const renderTabSpy = jest.spyOn(wrapper.instance(), 'renderTab');
    wrapper.instance().renderTab(props.steps[0], 1, props.steps[1].icon);
    expect(renderTabSpy).toHaveBeenCalled();
  });
});
