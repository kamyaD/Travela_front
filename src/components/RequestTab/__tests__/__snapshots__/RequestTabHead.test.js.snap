// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestTabHead /> should render properly 1`] = `
<div>
  <div
    className="new-request_title"
  >
    <div>
      <span
        className="new__request-title-text"
      >
        CREATE A NEW TRAVEL REQUEST
      </span>
    </div>
  </div>
  <div
    className="request__tab"
  >
    <div
      className="request__tab-card  complete"
      key="1"
    >
      <div
        className="mark"
      >
        <img
          alt=""
          src="mark.svg"
        />
      </div>
      <div
        className="tab-logo"
      >
        <personal
          color="#10A36D"
        />
      </div>
      <div
        className="tab-title"
      >
        Personal Information
      </div>
      <div
        className="foot-text"
      />
    </div>
    <div
      className="request__tab-card  complete"
      key="2"
    >
      <div
        className="mark"
      >
        <img
          alt=""
          src="mark.svg"
        />
      </div>
      <div
        className="tab-logo"
      >
        <tripDetails
          color="#10A36D"
        />
      </div>
      <div
        className="tab-title"
      >
        Trip Details
      </div>
      <div
        className="foot-text"
      />
    </div>
    <div
      className="request__tab-card current_request incomplete"
      key="3"
    >
      <div
        className="mark"
      >
        <img
          alt=""
          src="mark.svg"
        />
      </div>
      <div
        className="tab-logo"
      >
        <stipend
          color="#3359DB"
        />
      </div>
      <div
        className="tab-title"
      >
        Travel Stipends
      </div>
      <div
        className="foot-text"
      />
    </div>
    <div
      className="request__tab-card  incomplete"
      key="4"
    >
      <div
        className="mark"
      >
        <img
          alt=""
          src="mark.svg"
        />
      </div>
      <div
        className="tab-logo"
      >
        <checkList
          color="#3359DB"
        />
      </div>
      <div
        className="tab-title"
      >
        Travel Checklist
      </div>
      <div
        className="foot-text"
      />
    </div>
  </div>
</div>
`;
